### PR TITLE
Add io.github.tofudd.lanchat.terminal

### PR DESCRIPTION
**Pull request by tofudd**
### Please confirm your submission meets all the criteria

- [x] I have read the [App Requirements][reqs] and [App Maintenance][maint] pages.
- [x] My pull request follows the instructions at [App Submission][submission].
- [ ] I am using only the minimal set of permissions. *(If not, please explain each non-standard permission.)*
- [x] All assets referenced in the manifest are redistributable by any party.  If not, the unredistributable parts are using an extra-data source type.
- [x] I am an upstream contributor to the project. If not, I contacted upstream developers about submitting their software to Flathub.
- [x] I own the domain used in the application ID or the domain has a policy for delegating subdomains (e.g. GitHub, SourceForge).
- [x] Any additional patches or files have been submitted to the upstream projects concerned. *(If not, explain why.)*

[reqs]: https://github.com/flathub/flathub/wiki/App-Requirements
[maint]: https://github.com/flathub/flathub/wiki/App-Maintenance
[submission]: https://github.com/flathub/flathub/wiki/App-Submission

### Permissions:
- `network` for basic features and checking updates

### CODE REVIEW

**Comment by tofudd** [comment]
bot, build io.github.tofudd.lanchat.terminal

**Comment by tofudd** [comment]
bot, build io.github.tofudd.lanchat.terminal

**Comment by tofudd** [comment]
bot, build io.github.tofudd.lanchat.terminal

File: io.github.tofudd.lanchat.terminal.json [hunk 1]
```diff
@@ -0,0 +1,67 @@
...
+        "/usr/lib/sdk/dotnet5/bin/dotnet publish -c Release --source ../nuget-sources Lanchat.Terminal/Lanchat.Terminal.csproj --runtime $RUNTIME --self-contained true",
+        "cp -r --remove-destination /run/build/Lanchat/Lanchat.Terminal/bin/Release/net5.0/$RUNTIME/publish/ /app/bin/",
+        "install -D io.github.tofudd.lanchat.terminal.svg /app/share/icons/hicolor/128x128/apps/io.github.tofudd.lanchat.terminal.svg",
+        "install -D io.github.tofudd.lanchat.terminal.desktop /app/share/applications/io.github.tofudd.lanchat.terminal.desktop",
+        "install -D io.github.tofudd.lanchat.terminal.appdata.xml /app/share/metainfo/io.github.tofudd.lanchat.terminal.appdata.xml"
+      ],
+      "sources": [
+      {
+          "type": "file",
+          "path": "io.github.tofudd.lanchat.terminal.appdata.xml"
```
Comment (LeandroStanger) [inline io.github.tofudd.lanchat.terminal.json:48]: New appdata files should use metainfo instead. https://www.freedesktop.org/software/appstream/docs/chap-Quickstart.html

File: flathub.json [hunk 2]
```diff
@@ -0,0 +1,3 @@
+{
+  "only-arches": ["x86_64", "arm", "aarch64"]
+}
```
Comment (LeandroStanger) [inline flathub.json:3]:
```suggestion
{
  "publish-delay-hours": 1
}
```

Comment (tofudd) [reply flathub.json:3]: Why should `"only-arches"` be removed?

Comment (barthalion) [reply flathub.json:3]: Only x86_64 and aarch64 are supported these days. I don't understand the suggestion to use `publish-delay-hours` though, please just remove this file completely.

File: io.github.tofudd.lanchat.terminal.metainfo.xml [hunk 3]
```diff
@@ -0,0 +1,34 @@
+<?xml version="1.0" encoding="UTF-8"?>
```
Comment (barthalion) [inline io.github.tofudd.lanchat.terminal.metainfo.xml:1]: This file should be ideally maintained in your repository and not on Flathub; this way you can update it as part of the release workflow.

File: io.github.tofudd.lanchat.terminal.desktop [hunk 4]
```diff
@@ -0,0 +1,8 @@
+[Desktop Entry]
```
Comment (barthalion) [inline io.github.tofudd.lanchat.terminal.desktop:1]: This file should be ideally maintained in your repository and not on Flathub; this way you can update it as part of the release workflow.

File: io.github.tofudd.lanchat.terminal.svg [hunk 5]
```diff
@@ -0,0 +1,90 @@
+<?xml version="1.0" encoding="UTF-8" standalone="no"?>
```
Comment (barthalion) [inline io.github.tofudd.lanchat.terminal.svg:1]: Ditto.

File: io.github.tofudd.lanchat.terminal.metainfo.xml [hunk 6]
```diff
@@ -0,0 +1,34 @@
...
+  </content_rating>
+  <description>
+    <p>
+	Encrypted, P2P, local network chat.
+    </p>
+  </description>
+    <categories>
+    <category>Network</category>
+    <category>Chat</category>
+  </categories>
```
Comment (barthalion) [inline io.github.tofudd.lanchat.terminal.metainfo.xml:19]: Categories are imported from the desktop file.

File: io.github.tofudd.lanchat.terminal.json [hunk 7]
```diff
@@ -0,0 +1,67 @@
...
+  "runtime": "org.freedesktop.Platform",
+  "runtime-version": "20.08",
+  "sdk": "org.freedesktop.Sdk",
+  "sdk-extensions": [
+    "org.freedesktop.Sdk.Extension.dotnet5"
+  ],
+  "finish-args": [
+    "--share=network",
+    "--env=DOTNET_ROOT=/app/dotnet/lib",
+    "--filesystem=home"
```
Comment (barthalion) [inline io.github.tofudd.lanchat.terminal.json:12]: Ideally your app should respect XDG specification to store configuration files in XDG_CONFIG_HOME and data in XDG_DATA_HOME.

**Comment by barthalion** [comment]
bot, build io.github.tofudd.lanchat.terminal

File: io.github.tofudd.lanchat.terminal.metainfo.xml [hunk 8]
```diff
@@ -0,0 +1,34 @@
+<?xml version="1.0" encoding="UTF-8"?>
+<component type="console-application">
```
Comment (barthalion) [inline io.github.tofudd.lanchat.terminal.metainfo.xml]: Just a note – you may want to change it to desktop-application as we currently don't display console-apps on https://flathub.org

**Comment by barthalion** [comment]
One comment about metainfo file, otherwise LGTM. Feel free to address it after merge.

/merge

**Pull request closed without merging**